gdpr-cookie-consent/app/root.tsx

```diff
@@ -1,50 +1,47 @@
-import type { LoaderArgs, MetaFunction } from "@remix-run/node";
-import { json } from "@remix-run/node";
+import { LoaderFunctionArgs } from "@remix-run/node";
 import {
   Links,
-  LiveReload,
   Meta,
   Outlet,
   Scripts,
   ScrollRestoration,
+  json,
   useFetcher,
   useLoaderData,
+  useOutletContext,
 } from "@remix-run/react";
-import * as React from "react";
+import { useEffect } from "react";
 
-import { gdprConsent } from "~/cookies";
+import { gdprConsent } from "~/cookies.server";
 
-export const loader = async ({ request }: LoaderArgs) => {
+export const loader = async ({ request }: LoaderFunctionArgs) => {
   const cookieHeader = request.headers.get("Cookie");
   const cookie = (await gdprConsent.parse(cookieHeader)) || {};
   return json({ track: cookie.gdprConsent });
 };
 
-export const meta: MetaFunction = () => ({
-  charset: "utf-8",
-  title: "New Remix App",
-  viewport: "width=device-width,initial-scale=1",
-});
-
-export default function App() {
+export function Layout({ children }: { children: React.ReactNode }) {
   const { track } = useLoaderData<typeof loader>();
```

Comment:
@sergiodxa Little problem here, Layout shouldn't be calling useLoaderData() since it would fail when an error is thrown.

Tried calling useLoaderData() from <App /> and pass `track` in the Outlet (<Outlet context={{ track }} />) but useOutletContext() returns null in Layout (which seems normal since we're still in the same route, not in a descendant).

I'm curious how you would handle this? Dummy ErrorBoundary that doesn't use Layout ? A context ?

Reply:
Went with useRouteLoaderData() as a workaround. Still curious about your opinion here!